timemachine/cpp/src/centroid_restraint.hpp

```diff
@@ -14,6 +14,8 @@ template <typename RealType> class CentroidRestraint : public Potential {
     unsigned long long *d_centroid_a_;
     unsigned long long *d_centroid_b_;
 
+    __int128 *d_u_buffer_;
```

Comment:
nit/deboggle: it looks like `d_u_buffer_` is added as a member variable to most/all of the `Potential` subclasses? Would it make sense to declare in `Potential`?